### PR TITLE
[dagit] Virtualized schedules table

### PR DESCRIPTION
**Pull request by hellendag**
### Summary & Motivation

A first pass at a virtualized Schedules table to new Workspace page.

Screen Shot 2022-09-15 at 1 21 41 PM

### How I Tested These Changes

View Dagit with new workspace page flag enabled, click Workspace, then Schedules. Verify rendering of virtualized table, incuding fanout of individual row queries.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9706** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9706" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9706?utm_source=stack-comment).

File: js_modules/dagit/packages/core/src/workspace/WorkspaceTabs.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ export const WorkspaceTabs = <TData extends Record<string, any>>(props: Props<TD
     <Box flex={{direction: 'row', justifyContent: 'space-between', alignItems: 'flex-end'}}>
       <Tabs selectedTabId={tab}>
         <TabLink id="jobs" title="Jobs" to="/workspace/jobs" />
-        {/* <TabLink id="schedules" title="Schedules" to="/workspace/schedules" />
-        <TabLink id="sensors" title="Sensors" to="/workspace/sensors" />
+        <TabLink id="schedules" title="Schedules" to="/workspace/schedules" />
+        {/* <TabLink id="sensors" title="Sensors" to="/workspace/sensors" />
```
Comment (prha) [inline js_modules/dagit/packages/core/src/workspace/WorkspaceTabs.tsx:22]: should this all be visible?

File: js_modules/dagit/packages/core/src/workspace/VirtualizedScheduleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,336 @@
+import {gql, useLazyQuery} from '@apollo/client';
+import {Box, Button, Caption, Colors, Icon, Menu, Popover, Tag, Tooltip} from '@dagster-io/ui';
+import {useVirtualizer} from '@tanstack/react-virtual';
```
Comment (prha) [inline js_modules/dagit/packages/core/src/workspace/VirtualizedScheduleTable.tsx:3]: this looks cool.  is the plan to move everything off of `react-virtualized`?

**Review by prha — APPROVED** [review]

**Review by bengotow — APPROVED** [review]
This looks good to me - definitely want to check the commented line that Phil flagged before we merge. I vaguely wish we could move a bit more of the logic involved in using a virutalized table into the `ui/VirtualizedTable.tsx` file, but I don't see a great way of doing it without creating abstractions to hide abstractions, which seems like overkill.

File: js_modules/dagit/packages/core/src/workspace/WorkspaceTabs.tsx (hunk 1 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/workspace/WorkspaceTabs.tsx:22]: I'm exposing them in the tabs as I add the routes. The feature is flagged anyway, so no one will see this page unless they turn the flag on, and the routes referenced in the commented tabs don't exist yet.

File: js_modules/dagit/packages/core/src/workspace/VirtualizedScheduleTable.tsx (hunk 2 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/workspace/VirtualizedScheduleTable.tsx:3]: I think that may make sense with `react-virtualized` now looking like it's in maintenance mode.

**Comment by hellendag** [comment]
> I vaguely wish we could move a bit more of the logic involved in using a virutalized table into the ui/VirtualizedTable.tsx file, but I don't see a great way of doing it without creating abstractions to hide abstractions, which seems like overkill.

Yeah, my goal here is to avoid overabstracting too soon. These views are all pretty similar, but I wanted to put them together first and then pull out the shared pieces.

**Pull request merged**